src/rocksdb.rs

```diff
@@ -39,16 +40,134 @@ pub struct ReadOptions {
     inner: rocksdb_ffi::RocksDBReadOptions,
 }
 
+pub struct Snapshot<'a> {
+    db: &'a RocksDB,
+    inner: rocksdb_ffi::RocksDBSnapshot,
+}
+
+pub struct DBIterator {
+//TODO: should have a reference to DB to enforce scope, but it's trickier than I thought to add
+    inner: rocksdb_ffi::RocksDBIterator,
+    direction: Direction,
+    just_seeked: bool
+}
+
+pub enum Direction {
+  forward, reverse
+}
```

Comment:
Makes sense.  So, the downside to implementing DEI is that you need to have 2 internal RocksDB iterators, and because they exhaust the same range each next / next_back call needs to ensure that the two ends don't move past each other.  I think the upside is that by implementing DEI it will present a more idiomatic interface, and you get the rev() method for free, but I'm not sure if this is worth it or not.  I don't have a super strong opinion on it, but we may want to reconsider it in the future if it's confusing to people.